### ISSUE TITLE
`isSupportedVersion` should be based on Bee API versions

### ISSUE THREAD

**Issue by nugaon** [ethersphere/bee-js#463]
https://github.com/ethersphere/bee-js/pull/459#issuecomment-978056428

**Comment by nugaon** [comment]
maybe would be the best if `isSupportedVersion` returned 
```ts
{
  beeVersion: { supported: boolean, supportedVersion: string }, 
  beeApiVersion: { supported: boolean, supportedVersion: string }, 
  beeDebugApiVersion: { supported: boolean, supportedVersion: string } 
}
``` 

where obviously `suppertedVersion` comes from the package.json, and `supported` has some logic how we decide on the library supports the version fetched from the `health` EP or not (such as at semantic versioning on major changes it does not support).

**Comment by AuHau** [comment]
This was closed in the `3.3.0` release.
